api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1016,6 +1027,18 @@ message Cluster {
   // [#extension-category: envoy.network.dns_resolver]
   core.v3.TypedExtensionConfig typed_dns_resolver_config = 55;
 
+  // Optionally configures on-demand DNS resolution configuration. Note that this field controls the behavior of the cluster,
+  // not the DNS resolver itself (use
+  // :ref:`typed_dns_resolver_config <envoy_v3_api_field_config.cluster.v3.Cluster.typed_dns_resolver_config>`
+  // for those use-cases). At present there are 2 modes: DISABLED and DELAYED. DISABLED mode persists the background,
+  // asynchronous DNS resolution behavior that, at time of writing, is the default for STRICT_DNS and LOGICAL_DNS clusters.
+  // DELAYED mode will delay async DNS resolution until the first connection is made to the cluster in question.
+  // Once DNS resolution is complete, async resolution will continue periodically in the background as usual. Note that the latency of
+  // the DNS resolution will affect the first connection's RTT. Consequently, failure to resolve DNS before the connection's idle timeout
```

Comment:
```suggestion
  // the DNS resolution will affect the first connection's RTT. Consequently, failure to resolve DNS before the connect timeout
```

DId you mean connect timeout? If not, can you explain what you mean?

Reply:
Ah good catch; I get all the time outs mixed up sometimes 🙂 will fix